app/models/chargeback.rb

```diff
@@ -146,11 +144,9 @@ def get_rates(perf)
       end
   end
 
-  def self.calculate_costs(metric_rollup_records, rates, hours_in_interval)
-    calculated_costs = {}
-
-    chargeback_fields_present                = metric_rollup_records.count(&:chargeback_fields_present?)
-    calculated_costs['fixed_compute_metric'] = chargeback_fields_present if chargeback_fields_present
+  def calculate_costs(metric_rollup_records, rates, hours_in_interval)
+    chargeback_fields_present = metric_rollup_records.count(&:chargeback_fields_present?)
+    self.fixed_compute_metric = chargeback_fields_present if chargeback_fields_present
```

Comment:
Nod.

Although, I am not bringing the fragility here. I only make changes, that make this easier to spot.

Previously, we have build a hash (was stored in `data[key]` or `calculated_costs`), this hash had various keys not defined in `Chargeback` parent class. Later we passed this hash to `new()` and hoped for the best. We then modified the code to assume some of the keys are always present, like `fixed_compute_metric`. Perhaps we can later move cols that are always needed to the `Chargeback` parent class?

I have noticed, there are other parts of `Chargeback` class that rely on its descendants to define certain things. For example, `get_keys_and_extra_fields`, `get_rate_parents`, `report_col_options`, `report_static_cols`, `where_clause`, and perhaps more. Thus, this pattern is common in this part of the code.